zerver/actions/channel_folders.py

```diff
@@ -42,6 +48,18 @@ def check_add_channel_folder(
     return channel_folder
 
 
+@transaction.atomic(durable=True)
+def try_reorder_realm_channel_folders(realm: Realm, order: Iterable[int]) -> None:
+    order_mapping = {_[1]: _[0] for _ in enumerate(order)}
+    channel_folders = ChannelFolder.objects.filter(realm=realm)
+    for channel_folder in channel_folders:
+        if channel_folder.id not in order_mapping:
+            raise JsonableError(_("Invalid order mapping."))
+    for channel_folder in channel_folders:
+        channel_folder.order = order_mapping[channel_folder.id]
+        channel_folder.save(update_fields=["order"])
```

Comment:
We should make this do a `bulk_update`; there's no real cost, and if anyone has 50 channel folders, it'll make a significant difference performance-wise.

We can also skip updating any folders where the `order` value didn't change.